quaireaux/math/src/tests/sha256_test.cairo

```diff
@@ -0,0 +1,410 @@
+use quaireaux_math::sha256;
+use array::ArrayTrait;
+use debug::PrintTrait;
+
+#[test]
+#[available_gas(2000000000)]
+fn sha256_empty_test() {
+    let mut input = ArrayTrait::<u8>::new();
+    let result = sha256::sha256(input);
+
+    // result should be 0xE3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855
+    assert(result.len() == 32, 'invalid result length');
+    assert(*result[0] == 0xE3, 'invalid result');
+    assert(*result[1] == 0xB0, 'invalid result');
+    assert(*result[2] == 0xC4, 'invalid result');
+    assert(*result[3] == 0x42, 'invalid result');
+    assert(*result[4] == 0x98, 'invalid result');
+    assert(*result[5] == 0xFC, 'invalid result');
+    assert(*result[6] == 0x1C, 'invalid result');
+    assert(*result[7] == 0x14, 'invalid result');
+    assert(*result[8] == 0x9A, 'invalid result');
+    assert(*result[9] == 0xFB, 'invalid result');
+    assert(*result[10] == 0xF4, 'invalid result');
+    assert(*result[11] == 0xC8, 'invalid result');
+    assert(*result[12] == 0x99, 'invalid result');
+    assert(*result[13] == 0x6F, 'invalid result');
+    assert(*result[14] == 0xB9, 'invalid result');
+    assert(*result[15] == 0x24, 'invalid result');
+    assert(*result[16] == 0x27, 'invalid result');
+    assert(*result[17] == 0xAE, 'invalid result');
+    assert(*result[18] == 0x41, 'invalid result');
+    assert(*result[19] == 0xE4, 'invalid result');
+    assert(*result[20] == 0x64, 'invalid result');
+    assert(*result[21] == 0x9B, 'invalid result');
+    assert(*result[22] == 0x93, 'invalid result');
+    assert(*result[23] == 0x4C, 'invalid result');
+    assert(*result[24] == 0xA4, 'invalid result');
+    assert(*result[25] == 0x95, 'invalid result');
+    assert(*result[26] == 0x99, 'invalid result');
+    assert(*result[27] == 0x1B, 'invalid result');
+    assert(*result[28] == 0x78, 'invalid result');
+    assert(*result[29] == 0x52, 'invalid result');
+    assert(*result[30] == 0xB8, 'invalid result');
+    assert(*result[31] == 0x55, 'invalid result');
+}
+
+#[test]
+#[available_gas(2000000000)]
+fn sha256_lorem_ipsum_test() {
+    // Lorem ipsum, or lsipsum as it is sometimes known, is dummy text used in laying out print, graphic or web designs. The passage is attributed to an unknown typesetter in the 15th century who is thought to have scrambled parts of Cicero's De Finibus Bonorum et Malorum for use in a type specimen book. It usually begins with
```

Comment:
If you could make this comment on several lines, it could be handy to avoid having having a horizontal scroll bar :)